### PR TITLE
feat: add oldest_unconfirmed_vsc query 

### PR DESCRIPTION
**Pull request by mpoke**
## Description

Closes: #1263

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [x] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [x] Added `!` to the type prefix if the change is [state-machine breaking](https://github.com/cosmos/interchain-security/blob/main/RELEASES.md#breaking-changes)
* [x] Confirmed this PR does not introduce changes requiring state migrations, OR migration code has been added to consumer and/or provider modules
* [x] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
* [x] Provided a link to the relevant issue or specification
* [x] Followed the guidelines for [building SDK modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/00-intro.md)
* [x] Included the necessary unit and integration [tests](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#testing)
* [x] Added a changelog entry to `CHANGELOG.md`
* [x] Included comments for [documenting Go code](https://blog.golang.org/godoc)
* [ ] Updated the relevant documentation or specification
* [x] Reviewed "Files changed" and left comments if necessary 
* [ ] Confirmed all CI checks have passed
* [ ] If this PR is library API breaking, bump the go.mod version string of the repo, and follow through on a new major release

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

* [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed `!` the type prefix if the change is state-machine breaking
* [ ] confirmed this PR does not introduce changes requiring state migrations, OR confirmed migration code has been added to consumer and/or provider modules
* [ ] confirmed all author checklist items have been addressed
* [ ] reviewed state machine logic
* [ ] reviewed API design and naming
* [ ] reviewed documentation is accurate
* [ ] reviewed tests and test coverage

### CODE REVIEW

**Review by p-offtermatt — APPROVED** [review]
LGTM. The change in formatting for logging in ibc_module seems reasonable to have in this PR since it's just a tiny nit

File: x/ccv/provider/keeper/grpc_query.go [hunk 1]
```diff
@@ -221,3 +224,32 @@ func (k Keeper) QueryParams(c context.Context, _ *types.QueryParamsRequest) (*ty
...
+	}
+
+	if _, consumerRegistered := k.GetConsumerClientId(ctx, req.ChainId); !consumerRegistered {
+		return nil, status.Error(
+			codes.NotFound,
+			errorsmod.Wrap(types.ErrUnknownConsumerChainId, req.ChainId).Error(),
+		)
+	}
+
+	ts, found := k.GetFirstVscSendTimestamp(ctx, req.ChainId)
```
Comment (insumity) [inline x/ccv/provider/keeper/grpc_query.go:246]: ni: Could add a comment here on why `GetFirstVscSendTimestamp` returns the timestamp from the oldest unconfirmed `VSCPacket`, e.g., that everything under the timestamps are getting deleted when we handle `VSCMaturedPacket`s etc.

File: x/ccv/provider/keeper/grpc_query_test.go [hunk 2]
```diff
@@ -54,3 +55,36 @@ func TestQueryAllPairsValConAddrByConsumerChainID(t *testing.T) {
...
+
+	// Request with chainId is empty
+	_, err = pk.QueryOldestUnconfirmVsc(ctx, &types.QueryOldestUnconfirmVscRequest{})
+	require.Error(t, err)
+
+	// Request with chainId is invalid
+	_, err = pk.QueryOldestUnconfirmVsc(ctx, &types.QueryOldestUnconfirmVscRequest{ChainId: "invalidChainId"})
+	require.Error(t, err)
+
+	// Request is valid
```
Comment (insumity) [inline x/ccv/provider/keeper/grpc_query_test.go:82]: Might make sense to test with at least 2 `VscSendTimestamp`s here to verify that the oldest is returned (?)

File: proto/interchain_security/ccv/provider/v1/query.proto [hunk 3]
```diff
@@ -99,6 +99,13 @@ service Query {
       option (google.api.http).get =
     "/interchain_security/ccv/provider/params";
     }
+
+  // QueryOldestUnconfirmVsc returns the send timestamp of the oldest unconfirmed VSCPacket for a given chainID
+  rpc QueryOldestUnconfirmVsc(QueryOldestUnconfirmVscRequest)
```
Comment (insumity) [inline proto/interchain_security/ccv/provider/v1/query.proto:104]: Why not `QueryOldestUnconfirmedVsc`? Similarly for `QueryOldestUnconfirmVscRequest` and `QueryOldestUnconfirmVscResponse`.

**Review by insumity — APPROVED** [review]

**Pull request merged**